monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -168,53 +177,86 @@ def _create_cmd(self, train_params=None):
             for file in os.listdir(config_dir):
                 if not (file.endswith("yaml") or file.endswith("json")):
                     continue
-                if len(base_cmd) == 0:
-                    base_cmd += f"{train_py} run --config_file="
-                else:
-                    base_cmd += ","  # Python Fire does not accept space
+                base_cmd += f"{train_py} run --config_file=" if len(base_cmd) == 0 else ","
                 # Python Fire may be confused by single-quoted WindowsPath
                 config_yaml = Path(os.path.join(config_dir, file)).as_posix()
                 base_cmd += f"'{config_yaml}'"
-
-        if "CUDA_VISIBLE_DEVICES" in params:
-            devices = params.pop("CUDA_VISIBLE_DEVICES")
-            n_devices, devices_info = len(devices), ",".join([str(x) for x in devices])
-        else:
-            n_devices, devices_info = torch.cuda.device_count(), ""
-        if n_devices > 1:
-            cmd = f"torchrun --nnodes={1:d} --nproc_per_node={n_devices:d} "
+        cmd: str | None = self.device_setting["CMD_PREFIX"]  # type: ignore
+        # make sure cmd end with a space
+        if cmd is not None and not cmd.endswith(" "):
+            cmd += " "
+        if (
+            (int(self.device_setting["NUM_NODES"]) > 1 and self.device_setting["MN_START_METHOD"] == "bcprun")
+            or (int(self.device_setting["NUM_NODES"]) <= 1 and int(self.device_setting["n_devices"]) <= 1)
+        ):
+            cmd = "python " if cmd is None else cmd
+        elif int(self.device_setting["NUM_NODES"]) > 1:
+            raise NotImplementedError(
+                f"{self.device_setting['MN_START_METHOD']} is not supported yet."
+                "Try modify BundleAlgo._create_cmd for your cluster."
+            )
         else:
-            cmd = "python "  # TODO: which system python?
+            if cmd is None:
+                cmd = f"torchrun --nnodes={1:d} --nproc_per_node={self.device_setting['n_devices']:d} "
         cmd += base_cmd
         if params and isinstance(params, Mapping):
             for k, v in params.items():
                 cmd += f" --{k}={v}"
-        return cmd, devices_info
+        return cmd, ""
```

Comment:
kept this in order to be compatible with the existing algos  https://github.com/Project-MONAI/research-contributions/blob/8b10ad5b04d60835e1a2bd16a49abc378045b3f8/auto3dseg/algorithm_templates/dints/scripts/algo.py#L426-L428